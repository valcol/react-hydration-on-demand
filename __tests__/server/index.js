/**
 * @jest-environment node
 */

import React from "react";
import ReactDOMServer from "react-dom/server";
import withHydrationOnDemand from "../../src";

const Component = ({ label }) => <div className="label">{label}</div>;

test("Render correctly server side ", () => {
    const ComponentWithHydrationOnDemand = withHydrationOnDemand({
        on: [["scroll", () => document]]
    })(Component);

    const component = ReactDOMServer.renderToString(
        <ComponentWithHydrationOnDemand label="some content server side" />
    );
    expect(component).toMatchSnapshot();
});

test("Render correctly server side with custom wrapper props", () => {
    const ComponentWithHydrationOnDemand = withHydrationOnDemand()(Component);

    const component = ReactDOMServer.renderToString(
        <ComponentWithHydrationOnDemand
            wrapperProps={{
                className: "test-classname",
                style: { display: "contents" }
            }}
            label="some content server side"
        />
    );
    expect(component).toMatchSnapshot();
});

test("Render correctly server side with custom wrapper element", () => {
    const ComponentWithHydrationOnDemand = withHydrationOnDemand()(Component);

    const component = ReactDOMServer.renderToString(
        <ComponentWithHydrationOnDemand
            wrapperProps={{
                as: 'div',
                className: "test-classname",
                style: { display: "contents" }
            }}
            label="some content server side"
        />
    );
    expect(component).toMatchSnapshot();
});
