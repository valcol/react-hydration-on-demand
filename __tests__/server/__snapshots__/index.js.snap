// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly server side  1`] = `"<section data-hydration-on-demand=\\"true\\"><div class=\\"label\\">some content server side</div></section>"`;

exports[`Render correctly server side with custom wrapper element 1`] = `"<div data-hydration-on-demand=\\"true\\" class=\\"test-classname\\" style=\\"display:contents\\"><div class=\\"label\\">some content server side</div></div>"`;

exports[`Render correctly server side with custom wrapper props 1`] = `"<section data-hydration-on-demand=\\"true\\" class=\\"test-classname\\" style=\\"display:contents\\"><div class=\\"label\\">some content server side</div></section>"`;
